src/utility.rs

```diff
@@ -247,6 +265,29 @@ pub(crate) fn get_root(path: &Path) -> Result<FuncSpace> {
     Ok(root)
 }
 
+// Get all spaces stating from root.
+// It does not contain the root
+pub(crate) fn get_spaces(root: &FuncSpace) -> Result<Vec<(&FuncSpace, String)>> {
+    let mut stack = vec![(root, "".to_string())];
+    let mut result = Vec::<(&FuncSpace, String)>::new();
+    while let Some((space, path)) = stack.pop() {
+        for s in &space.spaces {
+            let p = format!(
+                "{}/{} ({},{})",
+                path,
+                s.name.as_ref().ok_or(Error::PathConversionError())?,
+                s.start_line,
+                s.end_line
+            );
+            stack.push((s, p.clone()));
```

Comment:
Question

To avoid a clone, can we create the path iteratively at each `for` step and then reset it in the `while_let`? In this way we don't need the `stack` vector, but only the `result` one. Because a clone of the space is really memory-hungry